src/core/models.py

```diff
@@ -454,3 +464,38 @@ def extra_user_roles(self) -> str:
         When `admin` is set, the Kafka charm interprets this as a new super.user.
         """
         return self.relation_data.get("extra-user-roles", "")
+
+
+class OAuth:
```

Comment:
**suggestion** maybe one could use the data class already provided by the charm lib and its parsing [here](https://github.com/canonical/kafka-operator/pull/168/files#diff-747d8e831b4929f2bce44463876e22478664105f47ed7b3f05ca14182b26a250R320)

Reply:
TBF I like that it's in line with the rest of the models here.

Reply:
@zmraul I did that at some point, once alone and once with Marc, but the `RelationState` is somehow tied to `Data` that comes from `data_platfrom_libs`, and the code was becoming more and more complicated for no good reason.

Reply:
Yeah it was weird. Having a duck-typed model is OK I reckon, we may need more in the future when integrating other non-DP charm libs.

Reply:
got it, thanks

Reply:
I am letting this like this for now, due to split decision :smile:

Reply:
It was a suggestion, which is not meant to be blocking. Resolving now (and feel free to resolve suggestion that stays as such, hence not blocking)